util/pkg/vfs/vaultcontext.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vfs
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"k8s.io/klog"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	vault "github.com/hashicorp/vault/api"
+)
+
+func newVaultClient(scheme string, host string, port string) (*vault.Client, error) {
+	addr := scheme + host + ":" + port
+	config := &vault.Config{
+		Address:    addr,
+		HttpClient: &http.Client{},
+	}
+	client, err := vault.NewClient(config)
+	if err != nil {
+		return nil, err
+	}
+
+	token := os.Getenv("VAULT_TOKEN")
+	if token == "" {
+		token, err = awsAuth(client, host)
+		if err != nil {
+			klog.Infof("Error authenticating vault to AWS: %v", err)
```

Comment:
Shouldn't we return err here, not below? And we shouldn't both log and return an error.

Reply:
We should return here. This was a leftover of an attempt at chaining with other auth methods.
But I think we need to move fi.Cloud to another package in order to do so, so I can use it here without running into import cycles.